File: ios/OctopusTopAppBarConfig.swift
import Foundation

/// Parsed representation of the JS `topAppBar` config.
struct OctopusTopAppBarConfig {
  /// "logo", "text", or nil (omitted → logo default).
  let titleType: String?
  /// Custom title text when `titleType` is "text".
  let titleText: String?
  /// true → centered title; false → leading.
  let centered: Bool
  /// true → primary color as nav-bar background (iOS 16+).
  let coloredBackground: Bool
}

File: android/src/main/java/com/octopuscommunity/octopusreactnativesdk/OctopusTopAppBarConfig.kt
package com.octopuscommunity.octopusreactnativesdk

/**
 * Parsed representation of the JS `topAppBar` config.
 *
 * @property titleType "logo", "text", or null (omitted → logo default)
 * @property titleText custom title text when [titleType] is "text"
 * @property centered true → centered title; false → leading
 * @property coloredBackground true → primary color as nav-bar background
 */
data class OctopusTopAppBarConfig(
  val titleType: String?,
  val titleText: String?,
  val centered: Boolean,
  val coloredBackground: Boolean
)
